src/dependency-registry.ts

```diff
@@ -0,0 +1,388 @@
+    for (let i = scriptCount; i < prevScriptCount; i++) {
+      const dependencyScript = DependencyScript.load(
+        generateDependencyScriptId(
+          event.params._dependencyType.toString(),
+          BigInt.fromI32(i)
+        )
+      );
+      if (dependencyScript) {
+        store.remove("DependencyScript", dependencyScript.id);
+      }
+    }
+  }
+
+  let scripts: string[] = [];
+  for (let i = 0; i < scriptCount; i++) {
+    let script = dependencyRegistryContract.getDependencyScriptAtIndex(
+      event.params._dependencyType,
+      BigInt.fromI32(i)
+    );
+
+    let dependencyScriptIndex = BigInt.fromI32(i);
+    let dependencyScript = new DependencyScript(
+      generateDependencyScriptId(dependencyType, dependencyScriptIndex)
+    );
+    dependencyScript.index = dependencyScriptIndex;
+    dependencyScript.dependency = dependencyType;
+    dependencyScript.script = script;
+    dependencyScript.save();
+
+    if (script) {
+      scripts.push(script);
+    }
+  }
+
+  const script = scripts.join("");
+
+  dependency.script = script;
+  dependency.scriptCount = scriptCount;
+
+  dependency.save();
+}
+
+export function handleSupportedCoreContractAdded(
+  event: SupportedCoreContractAdded
+): void {
+  const coreContract = Contract.load(
+    event.params._coreContractAddress.toHexString()
+  );
+
+  if (!coreContract) {
+    return;
+  }
+
+  coreContract.dependencyRegistry = event.address;
+  coreContract.updatedAt = event.block.timestamp;
+  coreContract.save();
+}
+
+export function handleSupportedCoreContractRemoved(
+  event: SupportedCoreContractRemoved
+): void {
+  const coreContract = Contract.load(
+    event.params._coreContractAddress.toHexString()
+  );
+
+  if (!coreContract) {
+    return;
+  }
+
+  coreContract.dependencyRegistry = null;
+  coreContract.updatedAt = event.block.timestamp;
+  coreContract.save();
+}
+
+export function handleProjectDependencyTypeOverrideAdded(
+  event: ProjectDependencyTypeOverrideAdded
+): void {
+  const projectId = generateContractSpecificId(
+    event.params._coreContractAddress,
+    event.params._projectId
+  );
+  const project = Project.load(projectId);
+  if (!project) {
+    return;
+  }
+
+  project.scriptTypeAndVersion = event.params._dependencyType.toString();
+  project.updatedAt = event.block.timestamp;
+  project.save();
+}
+
+export function handleProjectDependencyTypeOverrideRemoved(
+  event: ProjectDependencyTypeOverrideRemoved
+): void {
+  const projectId = generateContractSpecificId(
+    event.params._coreContractAddress,
+    event.params._projectId
+  );
+  const project = Project.load(projectId);
+  if (!project) {
+    return;
+  }
+
+  const coreContract = IDependencyRegistryCompatibleV0.bind(
+    event.params._coreContractAddress
+  );
+
+  const scriptDetailsResult = coreContract.try_projectScriptDetails(
+    event.params._projectId
+  );
+  if (scriptDetailsResult.reverted) {
+    project.scriptTypeAndVersion = null;
```

Comment:
just double-clicking on this - looks like we are good (on both smart contract and here) with not integrating directly with any pre-V3 core contracts, and instead requiring overrides to be set on the dependency registry 👍

Reply:
yupyup that's the intention